### DISCUSSION TITLE
How to use Pwncat in Docker

### DISCUSSION THREAD

**Discussion by MattPlayGamez** [calebstewart/pwncat#254] [Q&A]
Hi,
I just installed Pwncat on docker(windows). If I run `docker run -P -it pwncatimage bash `.. It doesn't publish the ports when using
 `listen --platform windows 8080`. Does any1 know how to fix this issue.

PS: I've locally build the image. 99% is the same. Below you can see the file
[Pastebin](https://pastebin.com/Ba6vywkb) | https://pastebin.com/Ba6vywkb.

Thanks in advance!

**Comment by remy415** [comment]
Matt, based on the documentation the --platform windows arg is for when your -target- is Windows based. Since you're loading it in a Docker container, it is being run in a Linux environment (even though Docker is being run on windows -- that is mostly transparent the containers). Confirm your "target" is a Windows machine, and if it isn't then remove the "--platform windows" arg.

Additionally, you need to forward the port from the host Windows machine to the container's 8080 port.  When the container is listening on a port for a callback, it isn't going to show up on the host system by default. Containers usually get their own IP addresses, so examine your networking setup and confirm the port is both reachable and forwarded on the host system. Windows Firewall will likely block incoming connection requests by default, and since you have it configured to listen for a callback you will need to reconfigure Windows Firewall or turn it off. Note: both of those options have their own security considerations, don't turn off or gut your firewall unless you know what you're doing.
